### PR TITLE
test: add test for decoding PSBT with per-input preimage types

### PR DESCRIPTION
**Pull request by theStack**
This PR adds missing test coverage for the `decodepsbt` RPC in the case that a PSBT with on of the per-input preimage types (`PSBT_IN_RIPEMD160`, `PSBT_IN_SHA256`, `PSBT_IN_HASH160`, `PSBT_IN_HASH256`; see [BIP 174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#Specification)) is passed. As preparation, the first four commits move the already existing helpers for (de)serialization of PSBTs and PSBTMaps from the signet miner to the test framework (in a new module `psbt.py`), which should be quite useful for further tests to easily create PSBTs.

### CODE REVIEW

**Comment by achow101** [comment]
ACK 4dc573e78976dc2728464714a589f4bc2a07e37d

**Review by brunoerg — APPROVED** [review]
crACK 4dc573e78976dc2728464714a589f4bc2a07e37d

**Comment by instagibbs** [comment]
concept ACK, moving the PSBT utility functions into the testing framework proper is something I wanted regardless

**Comment by theStack** [comment]
Rebased on master (due to trivial merge conflict in rpc_psbt.py after commit d2ed97656bba050051cfc677f1fa7eb3fc633f7d, #25590), should be easy to re-review.

**Review by instagibbs — COMMENTED** [review]
looks good aside from one question of mine

File: test/functional/rpc_psbt.py [hunk 1]
```diff
@@ -775,5 +788,37 @@ def test_psbt_input_keys(psbt_input, keys):
             self.nodes[0].sendrawtransaction(rawtx)
             self.generate(self.nodes[0], 1)
 
+        self.log.info("Test decoding PSBT with per-input preimage types")
+        # note that the decodepsbt RPC doesn't check whether preimages and hashes match
+        hash_ripemd160, preimage_ripemd160 = os.urandom(20), os.urandom(50)
```
[the remark this answers is not in the document]
Comment (theStack) [reply test/functional/rpc_psbt.py]: Good point, I agree that seed-determined randomness is preferred here. Moved the `random_bytes` helper (which uses `random.getrandbits`) from the taproot test to the utils library and used that instead. Note that Python 3.9 offers `random.randbytes()`, but unfortunately we have Python 3.6 as minimum requirement.

**Comment by achow101** [comment]
ACK 71a751f6c3e8912e1b1cfe388e593309d210e576

**Pull request merged**